docs/src/pages/components/textfield/GetFieldControls.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { DetailedHTMLProps, InputHTMLAttributes } from 'react';
+import { CheckboxField, Flex, TextField } from '@aws-amplify/ui-react';
+
+type FieldValue = DetailedHTMLProps<
+  InputHTMLAttributes<HTMLInputElement>,
+  HTMLInputElement
+>['value'];
+type FieldSetter = (event: any) => void;
+type FieldName = string;
+type ControlType = 'checkbox' | 'text' | string;
+export type FieldControl = [FieldValue, FieldSetter, FieldName, ControlType];
+
+export interface GetFieldControlsProps {
+  typeName: string;
+  fields: FieldControl[];
+}
+
+export const GetFieldControls = ({ fields }: GetFieldControlsProps) => {
```

Comment:
I do saw a few type errors that cannot be resolved out there, but they don't affect the docs functioning, so I didn't dedicate much time to overhaul this component. I like to make things generic, but sometimes it could also become complicated and less flexible. That's something we should think about before doing that